lib/improver/utilities/cube_manipulation.py

```diff
@@ -157,51 +152,435 @@ def _strip_var_names(cubes):
+                                    long_name='model_realization'))
+                            cube.add_aux_coord(new_model_real_coord,
+                                               data_dims=data_dims)
+                        else:
+                            new_model_real_coord = (
+                                build_coordinate(
+                                    [model_id_val],
+                                    long_name='model_realization'))
+                            cube.add_aux_coord(new_model_real_coord)
+                            new_realization_coord = (
+                                build_coordinate(
+                                    [0],
+                                    template_coord=realization_coord))
+                            cube.add_aux_coord(new_realization_coord)
+
+                # if mismatching is a dimension coord add to list to
+                # slice over.
+                if unmatching_coords[i][key]['data_dims'] is not None:
+                    slice_over_keys.append(key)
+                if unmatching_coords[i][key]['aux_dims'] is not None:
+                    slice_over_keys.append(key)
+
+            if len(slice_over_keys) > 0:
+                for slice_cube in cube.slices_over(slice_over_keys):
+                    cubelist.append(slice_cube)
+            else:
+                cubelist.append(cube)
+
+    return cubelist
+
+
+def compare_attributes(cubes):
+    """
+    Function to compare attributes of cubes
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to compare (must be more than 1)
+
+    Returns:
+        unmatching_attributes : List
+            List of dictionaries of unmatching attributes
+
+    Warns:
+        Warning: If only a single cube is supplied
+    """
+    unmatching_attributes = []
+    if len(cubes) == 1:
+        msg = ('Only a single cube so no differences will be found ')
+        warnings.warn(msg)
+    else:
+        common_keys = cubes[0].attributes.keys()
+        for cube in cubes[1:]:
+            cube_keys = cube.attributes.keys()
+            common_keys = [
+                key for key in common_keys
+                if (key in cube_keys and
+                    np.all(cube.attributes[key] == cubes[0].attributes[key]))]
+
+        for i, cube in enumerate(cubes):
+            unmatching_attributes.append(dict())
+            for key in cube.attributes.keys():
+                if key not in common_keys:
+                    unmatching_attributes[i].update({key:
+                                                     cube.attributes[key]})
+    return unmatching_attributes
+
+
+def compare_coords(cubes):
+    """
+    Function to compare the coordinates of the cubes
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to compare (must be more than 1)
+
+    Returns:
+        unmatching_coords : List
+            List of dictionaries of unmatching coordinates
```

Comment:
Could you extend this docstring to say that the number of `unmatching_coords` that you're returning is equal to the number of cubes that are input?

Reply:
Done